rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake

```diff
@@ -156,44 +148,23 @@ add_dependencies(
 )
 
 target_link_libraries(
-  ${_target_name_lib}
-  ${PythonExtra_LIBRARIES}
+  ${_target_name_lib} PRIVATE
+  Python3::NumPy
+  Python3::Python
 )
 target_include_directories(${_target_name_lib}
   PRIVATE
   ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_c
   ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_py
-  ${PythonExtra_INCLUDE_DIRS}
-)
-
-# Check if numpy is in the include path
-find_file(_numpy_h numpy/numpyconfig.h
-  PATHS ${PythonExtra_INCLUDE_DIRS}
 )
 
-if(APPLE OR WIN32 OR NOT _numpy_h)
-  # add include directory for numpy headers
-  set(_python_code
-    "import numpy"
-    "print(numpy.get_include())"
-  )
-  execute_process(
-    COMMAND "${PYTHON_EXECUTABLE}" "-c" "${_python_code}"
-    OUTPUT_VARIABLE _output
-    RESULT_VARIABLE _result
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-  )
-  if(NOT _result EQUAL 0)
-    message(FATAL_ERROR
-      "execute_process(${PYTHON_EXECUTABLE} -c '${_python_code}') returned "
-      "error code ${_result}")
-  endif()
-  message(STATUS "Using numpy include directory: ${_output}")
-  target_include_directories(${_target_name_lib} PUBLIC "${_output}")
+set(_extension_compile_flags "")
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(_extension_compile_flags -Wall -Wextra)
 endif()
 
 rosidl_get_typesupport_target(c_typesupport_target "${rosidl_generate_interfaces_TARGET}" "rosidl_typesupport_c")
-target_link_libraries(${_target_name_lib} ${c_typesupport_target})
+target_link_libraries(${_target_name_lib} PRIVATE ${c_typesupport_target})
```

Comment:
I can't quite figure it out in my mind, but setting the c type support library dependency as private seems like it could have some relationship to the Windows Debug test failures:

https://ci.ros2.org/job/ci_windows/20772/testReport/junit/(root)/projectroot/test_publisher_subscriber__rclpy__rmw_connextdds/

```
[test_subscriber-3] Traceback (most recent call last):
[test_subscriber-3]   File "C:\ci\ws\install\Lib\site-packages\rosidl_generator_py\import_type_support_impl.py", line 46, in import_type_support
[test_subscriber-3]     return importlib.import_module(module_name, package=pkg_name)
[test_subscriber-3]   File "C:\Python38\lib\importlib\__init__.py", line 127, in import_module
[test_subscriber-3]     return _bootstrap._gcd_import(name[level:], package, level)
[test_subscriber-3]   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
[test_subscriber-3]   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
[test_subscriber-3]   File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
[test_publisher-2]     check_is_valid_msg_type(msg_type)
[test_publisher-2]   File "C:\ci\ws\install\Lib\site-packages\rclpy\type_support.py", line 35, in check_is_valid_msg_type
[test_publisher-2]     check_for_type_support(msg_type)
[test_publisher-2]   File "C:\ci\ws\install\Lib\site-packages\rclpy\type_support.py", line 29, in check_for_type_support
[test_publisher-2]     msg_or_srv_type.__class__.__import_type_support__()
[test_publisher-2]   File "C:\ci\ws\install\Lib\site-packages\rcl_interfaces\msg\_parameter_event.py", line 36, in __import_type_support__
[test_publisher-2]     module = import_type_support('rcl_interfaces')
[test_publisher-2]   File "C:\ci\ws\install\Lib\site-packages\rosidl_generator_py\import_type_support_impl.py", line
```

Reply:
Or maybe the first place to look is to ask what is `import_type_support()` doing?

Reply:
I think we solved this with `set_target_properties(${_target_name} PROPERTIES DEBUG_POSTFIX "${PythonExtra_POSTFIX}")`.  I'm going to resolve this for now, and I'll keep an eye on the CI at https://github.com/ros2/python_cmake_module/pull/7#issuecomment-1912646583